File: 026/remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        length = len(nums)
        if length == 0:
            return 0

        i = 0
        for num in nums:
            if num != nums[i]:
                i += 1
                nums[i] = num

        return i + 1

File: 026/remove-duplicates-from-sorted-array.go
package main

import "fmt"

func removeDuplicates(nums []int) int {
	length := len(nums)

	if length == 0 {
		return 0
	}

	i := 0
	for _, num := range nums {
		if num != nums[i] {
			i++
			nums[i] = num
		}
	}
	return i + 1
}

func main() {
	// nums := []int{1, 1, 2}
	nums := []int{0, 0, 1, 1, 1, 2, 2, 3, 3, 4}
	fmt.Println(removeDuplicates(nums))
	fmt.Println(nums)
}
